### ISSUE TITLE
The Egress policy (dry-run) with the access level set to null still shows it as toggle-able in the UI

### ISSUE THREAD

**Issue by ptran32** [terraform-google-modules/terraform-google-vpc-service-controls#166]
### TL;DR

`egress_policies_dry_run` with  `access_levels = []` still toggle / enable access level on the Egress rule - but without any access level selected
<img width="1000" alt="Screenshot 2024-10-16 at 10 04 55 PM" src="https://github.com/user-attachments/assets/85a4707b-6d61-4c06-b5de-833cfb285ae6">

### Expected behavior

I expect the access level to be disabled and not toggle-able. Please refer to the screenshot below.
<img width="1032" alt="Screenshot 2024-10-16 at 10 05 07 PM" src="https://github.com/user-attachments/assets/2a8738e2-a3f1-4829-8538-c3cf4afc10c3">

### Observed behavior

I can see that the feature is enabled, but no access level is selected. This causes the Ingress rule to malfunction, as the Egress has an enabled access level but lacks a specific selected access level.

### Terraform Configuration

```hcl
egress_policies_dry_run = [
    {
      "from" = {
        "sources" = {
          resources = [
            "projects/xxxxxx"
          ],
          access_levels = [
          ]
        },
        "identity_type" = ""
        "identities"    = ["serviceAccount:sa@my-project.iam.gserviceaccount.com"]
      }
      "to" = {
        "resources" = [
          "projects/xxxx"
        ]
        "operations" = {
          "bigquery.googleapis.com" = {
            "methods" = [
              "BigQueryRead.CreateReadSession",
              "BigQueryStorage.CreateReadSession"
            ]
          }
        }
      }
    }
  ]
```

### Terraform Version

```sh
Terraform v1.8.5
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v5.37.0
```

**Comment by guicane** [comment]
I'm having the same issue and would like that this module supports this, this breaks a bunch of our configurations and things have to be done manually.

**Comment by ptran32** [comment]
I fixed the issue with those lines below, which is slightly different from the recent [commit](https://github.com/terraform-google-modules/terraform-google-vpc-service-controls/commit/6cc00f955da77b88e1e666588274721c145c58a7)

Enforced mode:
```
source_restriction = length(lookup(egress_policies.value["from"], "sources", {})["access_levels"]) > 0 ? "SOURCE_RESTRICTION_ENABLED" : null
```

Dry-run mode:
```
source_restriction = length(lookup(egress_policies_dry_run.value["from"], "sources", {})["access_levels"]) > 0 ? "SOURCE_RESTRICTION_ENABLED" : null
```


The rationale is that I encountered an issue because some of my Egress rules include access_levels, while others do not. (Terraform error: "cannot find a common base type for all elements.")

Sample pseudo code demonstrating Egress rules without and with access_levels.

```
      "from" = {
        "sources" = {
          access_levels = []
        },
      "from" = {
        "sources" = {
          access_levels = [module.my_access_level]
        },
```
